pkg/server/server.go

```diff
@@ -308,3 +315,23 @@ func mapConditions(conditions []metav1.Condition) []*pb.Condition {
 
 	return out
 }
+
+//Until the middleware is done this function will not be able to get the token and will fail
+func (s *applicationServer) Authenticate(ctx context.Context, r *pb.AuthenticateRequest) (*pb.AuthenticateResponse, error) {
+
+	if !strings.HasPrefix(github.DefaultDomain, r.ProviderName) &&
```

Comment:
Let's use `NormalizedRepoURL` here

Reply:
not sure I understand this 🤔, what I'm trying to do is just to make sure the provider name passed in is either `github` or `gitlab` using the constants from `go-git-providers` 🙂

Reply:
`NormalizedRepoURL` has some helper methods to return a type-safe git provider name: https://github.com/weaveworks/weave-gitops/blob/main/pkg/gitproviders/provider.go#L624

Reply:
Nevermind I misunderstood. Let's try using the typesafe provider names rather than looking at string prefixes